src/dmd/escape.d

```diff
@@ -2430,3 +2447,37 @@ private bool setUnsafeDIP1000(Scope* sc, bool gag, Loc loc, const(char)* msg, Ro
         return false;
     }
 }
+
+/***************************************
+ * Check that taking the address of `v` is `@safe`
+ *
+ * It's not possible to take the address of a scope variable, because `scope` only applies
+ * to the top level indirection.
+ *
+ * Params:
+ *     v = variable that a reference is created
+ *     e = expression that takes the referene
+ *     sc = used to obtain function / deprecated status
+ *     gag = don't print errors
+ * Returns:
+ *     true if taking the address of `v` is problematic because of the lack of transitive `scope`
+ */
+private bool checkScopeVarAddr(VarDeclaration v, Expression e, Scope* sc, bool gag)
+{
+    if (!e.type)
+        return false;
+
+    if ((v.storage_class & STC.temp) || !v.isScope())
+        return false;
+
+    Type t = e.type.baseElemOf();
+    if (t.ty == Tarray || t.ty == Tpointer)
+    {
+        if (!t.nextOf().toBasetype().hasPointers())
```

Comment:
is there any reason why these 2 ifs are not combined in a single one?

Reply:
The code is inspired by https://github.com/dlang/dmd/blob/f98a76ad2c24ef360c76eb30e5e018a1722cbb11/src/dmd/mtype.d#L4410-L4415 and modified until it looked like this. I can combine them into one if statement if you want.

Reply:
Done